pkg/controller/studyjob/utils.go

```diff
@@ -209,3 +210,88 @@ func contains(l []string, s string) bool {
 	}
 	return false
 }
+
+func validateHPJob(instance *katibv1alpha1.StudyJob) error {
+	log.Printf("Validation for the HP job")
+	if instance.Spec.ParameterConfigs == nil {
+		return fmt.Errorf("No Spec.ParameterConfigs specified")
+	}
+	err := validateParameterConfigs(instance.Spec.ParameterConfigs, jobTypeHP)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func validateNASJob(instance *katibv1alpha1.StudyJob) error {
+	log.Printf("Validation for the NAS job")
+	if instance.Spec.NasConfig == nil {
+		return fmt.Errorf("No Spec.NasConfig specified")
+	}
+
+	if reflect.DeepEqual(instance.Spec.NasConfig.GraphConfig, katibv1alpha1.GraphConfig{}) {
+		return fmt.Errorf("Missing GraphConfig in NasConfig: %v", instance.Spec.NasConfig)
+	}
+
+	if instance.Spec.NasConfig.GraphConfig.InputSize == nil {
+		return fmt.Errorf("Missing InputSize in NasConfig.GraphConfig: %v", instance.Spec.NasConfig.GraphConfig)
+	}
+
+	if instance.Spec.NasConfig.GraphConfig.OutputSize == nil {
+		return fmt.Errorf("Missing OutputSize in NasConfig.GraphConfig: %v", instance.Spec.NasConfig.GraphConfig)
+	}
+
+	if instance.Spec.NasConfig.GraphConfig.NumLayers == 0 {
```

Comment:
NumLayers is not always in need for all NAS (ENAS need it). I think we need validate NAS considering suggestion algorithm

Reply:
@hougangliu I think we can check these parameters exactly for Reinforcement Learning algorithm for now. For NumLayers and Step here: https://github.com/kubeflow/katib/blob/d407be8252034585bc770ced2de36eb39b72ddd6/pkg/controller/studyjob/utils.go#L287, we should validate only for RL algorithm right now. 
What do you think about it?